bl-kernel/helpers/filesystem.class.php

```diff
@@ -278,8 +278,10 @@ public static function getSize($fileOrDirectory) {
 		// Directories
 		if (file_exists($fileOrDirectory)) {
 		    $size = 0;
-		    foreach(new RecursiveIteratorIterator(new RecursiveDirectoryIterator($fileOrDirectory)) as $file){
-		        $size += $file->getSize();
+		    foreach(new RecursiveIteratorIterator(new RecursiveDirectoryIterator($fileOrDirectory, FilesystemIterator::SKIP_DOTS)) as $file){
+		    	if (file_exists($file)) {
```

Comment:
Why you need this condition here ? the `RecursiveIteratorIterator` is getting all the files inside, only the files that exists at the moment.

Reply:
file_exists() returns false on broken symlinks. It is to avoid Runtime exception

Reply:
The issue is you are returning `False` in a recursive function when is doing an SUM. If you `return 0` instead of `return False` you never going to get an exception, or you can check if the function returns False then do not SUM.

Reply:
I don't think the bug is related to returning `false.` It's related to `$file->getSize();` / `SplFileInfo::getSize`

From: https://www.php.net/manual/en/splfileinfo.getsize.php
Errors/Exceptions : A RuntimeException will be thrown if the file does not exist or an error occurs.

What is happening is, the user tried to upload svg in a post. User deleted the svg. A broken symlink was created. While looping, it is listing the broken symlink. It calls `->getSize();` on the symlink and an exception occurs. I added the file_exists() to ignore broken symlinks since it returns false when it comes across them according to docs here https://www.php.net/file_exists `Note: This function will return FALSE for symlinks pointing to non-existing files.`

Reply:
Ok perfect, I got it, thanks